sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/channel/DefaultChannel.java

```diff
@@ -375,19 +389,21 @@ private synchronized void handleSendingFailure(@NonNull final GroupState groupSt
     @Override
     public synchronized void enqueue(@NonNull Log log, @NonNull final String groupName) {
 
-        /* Check if disabled with discarding logs. */
-        if (mDiscardLogs) {
-            MobileCenterLog.warn(LOG_TAG, "Channel is disabled, log are discarded.");
-            return;
-        }
-
         /* Check group name is registered. */
         final GroupState groupState = mGroupStates.get(groupName);
         if (groupState == null) {
             MobileCenterLog.error(LOG_TAG, "Invalid group name:" + groupName);
             return;
         }
 
+        /* Check if disabled with discarding logs. */
+        if (mDiscardLogs) {
```

Comment:
That will make the callback flow consistent but the SDK already knows that it will fail anyway. Looks no better idea for this case, I'll just call it before `onSendingFailed`.